src/DotNetty.Buffers/ByteBufferUtil.cs

```diff
@@ -570,6 +570,56 @@ static void AppendHexDumpRowPrefix(StringBuilder dump, int row, int rowStartInde
             }
         }
 
+
+        /// <summary>
+        /// Encode the given {@link CharBuffer} using the given {@link Charset} into a new {@link ByteBuf} which
+        /// is allocated via the {@link ByteBufAllocator}.
+        /// </summary>
+        /// <param name="alloc">The {@link IByteBufferAllocator} to allocate {@link IByteBuffer}.</param>
+        /// <param name="src">src The {@link String} to encode.</param>
+        /// <param name="encoding">charset The specified {@link Encoding}</param>
+        public static IByteBuffer EncodeString(IByteBufferAllocator alloc, string src, Encoding encoding)
+        {
+            return EncodeString0(alloc, src, encoding, 0);
+        }
+
+        /// <summary>
+        /// Encode the given {@link CharBuffer} using the given {@link Charset} into a new {@link ByteBuf} which
+        /// is allocated via the {@link ByteBufAllocator}.
+        /// </summary>
+        /// <param name="alloc">The {@link IByteBufferAllocator} to allocate {@link IByteBuffer}.</param>
+        /// <param name="src">src The {@link String} to encode.</param>
+        /// <param name="encoding">charset The specified {@link Encoding}</param>
+        /// <param name="extraCapacity">the extra capacity to alloc except the space for decoding.</param>
+        public static IByteBuffer EncodeString(IByteBufferAllocator alloc, string src, Encoding encoding, int extraCapacity)
+        {
+            return EncodeString0(alloc, src, encoding, extraCapacity);
+        }
+
+        static IByteBuffer EncodeString0(IByteBufferAllocator alloc, string src, Encoding encoding, int extraCapacity)
+        {
+            var bytes = encoding.GetBytes(src);
```

Comment:
I have altered and started to test this code on a Echo example

**New version**

This method works fine, but I got strange behavior below

```c#
static IByteBuffer EncodeString0(IByteBufferAllocator alloc, string src, Encoding encoding, int extraCapacity)
        {
            int length = encoding.GetMaxByteCount(src.Length) + extraCapacity;
            var release = true;

            IByteBuffer dst = alloc.Buffer(length);
            Contract.Assert(dst.HasArray, "Operation expects allocator to operate array-based buffers.");

            try
            {
                encoding.GetBytes(src, 0, src.Length, dst.Array, dst.ArrayOffset);
                release = false;

                return dst;
            }
            finally
            {
                if (release)
                {
                    dst.Release();
                }
            }
        }
```

I replaced

```C#
context.WriteAsync(buffer);
```

on

```c#
var buf2 = ByteBufferUtil.EncodeString(context.Allocator, "Hello World" + Environment.NewLine, Encoding.UTF8);
context.WriteAsync(buf2);
```

And message don't send (or client don't handle it). No errors. I am see, that buf2 and buffer have same data in Array field. I confuse.

Unique differences that I found, it's array size. (buffer 4096, buf2 42)

Do you have any ideas, what I am missing? Thanks.